### PR TITLE
[REF] partner_credit_limit: Separate function check_limit creating check_payment_type

### PR DESCRIPTION
**Pull request by ysantiago**

### CODE REVIEW

File: partner_credit_limit/model/account_invoice.py [hunk 1]
```diff
@@ -10,10 +10,16 @@ class AccontInvoice(models.Model):
...
-    def check_limit_credit(self):
+    def check_payment_type(self):
         for invoice in self:
             if invoice.payment_term_id.payment_type != 'credit':
                 return True
+
+    @api.multi
+    def check_limit_credit(self):
+        for invoice in self:
+            if invoice.check_payment_type():
```
Comment (JulioSerna) [inline partner_credit_limit/model/account_invoice.py]: Could you adapt with filtered?

File: partner_credit_limit/model/sale.py [hunk 2]
```diff
@@ -9,10 +9,16 @@ class SaleOrder(models.Model):
...
-    def check_limit(self):
+    def check_payment_type(self):
         for so in self:
             if so.payment_term_id.payment_type != 'credit':
                 return True
+
+    @api.multi
+    def check_limit(self):
+        for so in self:
+            if so.check_payment_type():
```
Comment (JulioSerna) [inline partner_credit_limit/model/sale.py]: same here

File: partner_credit_limit/model/account_invoice.py (hunk 1 above)
Comment (ysantiago) [reply partner_credit_limit/model/account_invoice.py]: @JulioSerna the change it was made in [5b2026d](https://github.com/Vauxoo/addons-vauxoo/pull/1260/commits/5b2026df870cf2c7a70fd4e0ab3316451f194aff)

File: partner_credit_limit/model/sale.py (hunk 2 above)
Comment (ysantiago) [reply partner_credit_limit/model/sale.py]: @JulioSerna the same here

File: partner_credit_limit/model/account_invoice.py [hunk 3]
```diff
@@ -9,10 +9,16 @@ class AccontInvoice(models.Model):
...
+    def check_payment_type(self):
+        if self.filtered(lambda invoice:
+                         invoice.payment_term_id.payment_type != 'credit'):
+            return True
+
     @api.multi
     def check_limit_credit(self):
         for invoice in self:
-            if invoice.payment_term_id.payment_type != 'credit':
+            if invoice.check_payment_type():
```
Comment (JulioSerna) [inline partner_credit_limit/model/account_invoice.py]: is posible filtered into for?

File: partner_credit_limit/model/sale.py [hunk 4]
```diff
@@ -8,10 +8,16 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    @api.multi
+    def check_payment_type(self):
+        if self.filtered(lambda so:
```
Comment (JulioSerna) [inline partner_credit_limit/model/sale.py:13]: is posible error by expected singleton?

File: partner_credit_limit/model/sale.py [hunk 5]
```diff
@@ -8,10 +8,16 @@
...
+                         so.payment_term_id.payment_type != 'credit'):
+            return True
+
     @api.multi
     def check_limit(self):
         for so in self:
-            if so.payment_term_id.payment_type != 'credit':
+            if so.check_payment_type():
```
Comment (JulioSerna) [inline partner_credit_limit/model/sale.py]: could you use filtered into for?

File: partner_credit_limit/model/sale.py (hunk 4 above)
Comment (ysantiago) [reply partner_credit_limit/model/sale.py:13]: No, self can be one or more registers.

File: partner_credit_limit/model/account_invoice.py (hunk 3 above)
Comment (ysantiago) [reply partner_credit_limit/model/account_invoice.py]: I don't understand how can I use filtered here.

File: partner_credit_limit/model/sale.py (hunk 5 above)
Comment (ysantiago) [reply partner_credit_limit/model/sale.py]: I don't understand how can I use filtered here.

File: partner_credit_limit/model/account_invoice.py (hunk 3 above)
Comment (moylop260) [reply partner_credit_limit/model/account_invoice.py]: `self.filtered(lambda inv: inv.check_payment_type()):`

File: partner_credit_limit/model/sale.py (hunk 5 above)
Comment (moylop260) [reply partner_credit_limit/model/sale.py]: `self.filtered(lambda inv: inv.check_payment_type()):`

File: partner_credit_limit/model/account_invoice.py (hunk 3 above)
Comment (ysantiago) [reply partner_credit_limit/model/account_invoice.py]: Roger that. Change is [here](https://github.com/Vauxoo/addons-vauxoo/pull/1260/commits/4e6c2e3893fbf50e90b2f1bdf5646d1e97ff9f57#diff-b0121e38e8533ca630f1b3d6fc3e8d2aR19)

**Pull request merged**